draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -221,70 +215,68 @@
-      </t>
-    </section>
-    <section title="Provider Authorization Function" anchor="pair-validation">
+      <section title="ASPA Registration Recommendations" anchor="rec1">
+        <t>
+          An ASPA object showing only AS 0 as a provider AS is referred to as an AS0 ASPA.
+          A non-transparent Route Server AS (RS AS) is one that includes its AS number in the AS_PATH.
+          Registering as AS0 ASPA is a statement by the registering AS that it has no transit providers, and it is also not an RS-client at a non-transparent RS AS.
+          If that statement is true, then the AS MUST register an AS0 ASPA.
+        </t>
+        <t>
+          Normally, the Provider ASes of a CAS would be congruent for the address family combinations {AFI 1 (IPv4), SAFI 1} and {AFI 2 (IPv6), SAFI 1}.
+          Exceptions to this are expected to be rare.
+          In any case, the CAS MUST list the union of all Provider ASes applicable to the address family combinations stated above in the SPAS and MUST also include any non-transparent RS AS(es) at which it is an RS-client.
+          In the procedures for the AS path verification described in this document (<xref target="pair-validation"/>, <xref target="verif"/>), the SPAS is always considered to be uniformly applicable to {AFI 1 (IPv4), SAFI 1} and {AFI 2 (IPv6), SAFI 1}.
+        </t>
+        <t>
+          A compliant AS or Route Server AS (RS AS) MUST have an ASPA.
+          An RS AS SHOULD register an AS0 ASPA.
+          An AS SHOULD NOT have more than one ASPA.
+          Such a practice helps prevent race conditions during ASPA updates that might affect prefix propagation.
+          The software that provides hosting for ASPA records SHOULD support enforcement of this practice.
+          During a transition process between different certificate authority (CA) registries, the ASPA records SHOULD be kept identical in all relevant registries.
+        </t>
+        <t>
+          Normally, a VAP (see <xref target="rec1"/>) is not expected to contain both an AS 0 and other Provider ASes, but an unexpected presence of AS 0 has no influence on the AS path verification procedures (see <xref target="pair-validation"/>, <xref target="verif"/>).
+        </t>
+        <t>
+          Each of the two ASes in a mutual-transit pair MUST register its ASPA including the other AS in its SPAS.
+        </t>
+        <t>
+          The ASes on the boundary of an AS Confederation MUST register ASPAs using the Confederation's global ASN as the CAS.
+        </t>
+      </section>
+    <section title="Hop-Check Function" anchor="pair-validation">
       <t>
-        Let AS x and AS y represent two unique ASes.
-        A provider authorization function, authorized(AS x, AS y), checks if the ordered pair of ASNs, (AS x, AS y), has the property that AS y is an attested provider of AS x per VAP-SPAS of AS x.
```

Comment:
Please don't make it hop-based. It just a function with parameters that checks if AS x authorized AS y to propagate its routes. There is no other semantic.

Reply:
Whenever we use this function, we are always checking two neighboring ASes in an AS path which is a hop. I think this is not critical. We can let it remain mainly because (as I have said before) the WG and implementers are very used to this since they discussed the algorithms extensively and found hop(..) a convenient notation.

Reply:
The name of the function should always correlate with its meaning. This function checks if ordered pair is authorized in terms of ASPA, that's all. It will be zero problem for early implementers to change their code, but simplify the work for those who follow.

Reply:
Otherwise you will have to describe the term hop, which depends on the direction of processing.

Reply:
@mitradir  @cjeker  

Alexander: Thanks. I have inserted a sentence based on your suggestion of "the direction of processing".  Now it reads: 

   Let AS x and AS y represent two unique ASes.  A hop-check function,
   hop(AS x, AS y), checks if the ordered pair of ASNs, (AS x, AS y),
   has the property that AS y is an attested provider of AS x per the
   VAP of AS x.  It can be applied in either direction (right to left or
   the opposite) to a hop in an AS_PATH.